diff --git a/docroot/profiles/panopoly/modules/contrib/field_group/field_group.module b/docroot/profiles/panopoly/modules/contrib/field_group/field_group.module
old mode 100644
new mode 100755
index 1081e8c..716ea39
--- a/docroot/profiles/panopoly/modules/contrib/field_group/field_group.module
+++ b/docroot/profiles/panopoly/modules/contrib/field_group/field_group.module
@@ -1401,6 +1401,7 @@ function theme_horizontal_tabs($variables) {
   $element = $variables['element'];
   // Add required JavaScript and Stylesheet.
   $element['#attached']['library'][] = array('field_group', 'horizontal-tabs');
+  drupal_add_library('field_group', 'horizontal-tabs');
 
   $output = '<h2 class="element-invisible">' . (!empty($element['#title']) ? $element['#title'] : t('Horizontal Tabs')) . '</h2>';
   $output .= '<div class="horizontal-tabs-panes">' . $element['#children'] . '</div>';
