diff --git a/contrib/current_search/current_search.block.inc b/contrib/current_search/current_search.block.inc
index 1fe1151..6892ac1 100644
--- a/contrib/current_search/current_search.block.inc
+++ b/contrib/current_search/current_search.block.inc
@@ -75,6 +75,18 @@ function current_search_block_info() {
 }
 
 /**
+ * Implements hook_ctools_block_info().
+ *
+ * @see http://drupal.org/node/1669918
+ */
+function current_search_ctools_block_info($module, $delta, &$info) {
+  // Give the current search blocks it's own categories.
+  $info['category'] = t('Current Search Blocks');
+  // Allow blocks to be used before the search results in Panels.
+  $info['render last'] = TRUE;
+}
+
+/**
  * Returns the content for a facet based on the delta.
  */
 function current_search_block_view($delta = '') {
