--- panopoly_admin.module.bak	2014-06-23 17:28:55.175495431 +0530
+++ panopoly_admin.module	2014-06-23 17:29:12.034573431 +0530
@@ -113,7 +113,7 @@
   }
 
   // Improve the URL selection form
-  if (isset($form['#entity_type']) && $form['#entity_type'] == 'node' && isset($form['path']) && isset($form['path']['alias'])) {
+  /*if (isset($form['#entity_type']) && $form['#entity_type'] == 'node' && isset($form['path']) && isset($form['path']['alias'])) {
     unset($form['path']['#type']); // remove the fieldset
     unset($form['path']['alias']['#description']); 
     $form['path']['alias']['#title'] = '<strong>' . t('Permalink') . ':</strong> ';
@@ -125,7 +125,9 @@
       }
       $form['#submit'][] = 'panopoly_admin_pathauto_submit';
     }
-  }
+  }  
+  */
+  unset($form['path']['alias']);
   
   // Add clearfix to node body container to prevent clearing issue with
   // vertical tabs - @see https://drupal.org/node/1686500
