diff --git a/docroot/sites/all/modules/contrib/scald/modules/fields/mee/mee.module b/docroot/sites/all/modules/contrib/scald/modules/fields/mee/mee.module
index 43d91c8..7f0c8ff 100644
--- a/docroot/sites/all/modules/contrib/scald/modules/fields/mee/mee.module
+++ b/docroot/sites/all/modules/contrib/scald/modules/fields/mee/mee.module
@@ -707,7 +707,7 @@ function _mee_process_item_value(&$item, $entity_type, $entity, $field, $delta)
   }
 
   // If $item['mee'] does not hold anything, load default data into it.
-  if (empty($item['mee']) || !is_array($item['mee']['resource_manager'])) {
+  if (empty($item['mee']) || empty($item['mee']['resource_manager'])|| !is_array($item['mee']['resource_manager'])) {
     _mee_load_resources($entity_type, $entity, $field, $delta, $item);
   }
 
@@ -737,9 +737,9 @@ function _mee_load_resources($entity_type, $entity, $field, $delta, &$item) {
       ->condition('field', $field['field_name'])
       ->condition('delta', $delta)
       ->execute();
-  $item['mee']['resource_manager'] = array();
+  $item['mee'] = array('resource_manager' => array());
   $item['mee']['resource_manager'] = $result->fetchAllAssoc('atom_sid', PDO::FETCH_ASSOC);
-  $item['mee']['resource_manager'][0] = array('weight' => 0, 'required' => FALSE);
+  $item['mee']['resource_manager'] = array(0 => array('weight' => 0, 'required' => FALSE));
 }
 
 /**
