
--- a/plugins/facetapi/filter_exclude_specified_items.inc
+++ b/plugins/facetapi/filter_exclude_specified_items.inc
@@ -43,8 +43,14 @@ class FacetapiFilterExcludeItems extends FacetapiFilter {
           }
         }
       }
-      if (!$exclude) {
-        $filtered_build[$key] = $item;
+      if ($this->settings->settings['invert']) {
+        if ($exclude) {
+          $filtered_build[$key] = $item;
+        }
+      } else {
+        if (!$exclude) {
+          $filtered_build[$key] = $item;
+        }
       }
     }
 
@@ -67,6 +73,12 @@ class FacetapiFilterExcludeItems extends FacetapiFilter {
       '#description' => t('Interpret each exclude list item as a regular expression pattern.<br /><small>(Slashes are escaped automatically, patterns using a comma can be wrapped in "double quotes", and if such a pattern uses double quotes itself, just make them double-double-quotes (""))</small>.'),
       '#default_value' => $this->settings->settings['regex'],
     );
+    $form['invert'] = array(
+      '#title' => t('Invert - only list matched items'),
+      '#type' => 'checkbox',
+      '#description' => t('Instead of excluding items based on the pattern specified above, only matching items will be displayed.<br />'),
+      '#default_value' => $this->settings->settings['invert'],
+    );
   }
 
   /**
@@ -76,6 +88,7 @@ class FacetapiFilterExcludeItems extends FacetapiFilter {
     return array(
       'exclude' => '',
       'regex' => 0,
+      'invert' => 0,
     );
   }
 }
