--- a/securepages.module	2014-03-19 12:08:07.000000000 +0200
+++ b/securepages.module	2014-03-19 12:08:37.000000000 +0200
@@ -201,8 +201,8 @@
   else {
     // Setting the redirect headers manually allows them to be cached.
     drupal_add_http_header('Location', url($url['path'], $url));
-    drupal_add_http_header('Status', '302 Found');
-    print "302 Found";
+    drupal_add_http_header('Status', '301 Moved permanently');
+    print "301 Moved permanently";
 
     // Store the response in the page cache.
     if (variable_get('cache', 0) && ($cache = drupal_page_set_cache())) {
@@ -398,4 +398,4 @@
     }
     drupal_set_message(t('SP: ' . $text, $options), 'warning', FALSE);
   }
-}
\ No newline at end of file
+}
